### PR TITLE
Allow load empty style uri

### PR DESCRIPTION
**Pull request by Chaoba**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [ ] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog>Allow load empty style uri</changelog>`.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

### Summary of changes
In the past we can't load an empty style uri as it's not valid, we will only get a black screen and some error log.
This pr adds a check for empty uri and will load an empty json `{}` if the uri is empty so that users can load an empty uri and then add sources and layers.
 
The screenshot for ImageSourceActivity with empty style uri
image

### CODE REVIEW

File: sdk/src/main/java/com/mapbox/maps/MapboxMap.kt [hunk 1]
```diff
@@ -99,7 +101,11 @@ class MapboxMap internal constructor(
     onMapLoadErrorListener: OnMapLoadErrorListener? = null
   ) {
     initializeStyleLoad(onStyleLoaded, onMapLoadErrorListener)
-    nativeMapWeakRef.call { (this as StyleManagerInterface).styleURI = styleUri }
+    if (styleUri.isEmpty()) {
+      nativeMapWeakRef.call { (this as StyleManagerInterface).styleJSON = "{}" }
+    } else {
+      nativeMapWeakRef.call { (this as StyleManagerInterface).styleURI = styleUri }
+    }
```
Comment (yunikkk) [inline sdk/src/main/java/com/mapbox/maps/MapboxMap.kt:108]: nit : 
```
nativeMapWeakRef.call { (this as StyleManagerInterface).styleJSON = if ... else ...
```

Comment (Chaoba) [reply sdk/src/main/java/com/mapbox/maps/MapboxMap.kt:108]: One is `styleJSON` and the other is `styleURI`, so we can't apply the if-else block you suggested.
This is for users who want to create a style fully programmatically on the application itself. For more detail, you can refer to the linked ticket.

**Comment by kiryldz** [comment]
@Chaoba does it actually solve underlying issue?
will it work now if using Style DSL:
```
 mapView.getMapboxMap().loadStyle(
      styleExtension = style("") {
        +rasterSource("osm"){
          tileSet(tileSet)
          tileSize(256)
        }
        +rasterLayer(LAYER_ID, "osm") {

        }
      }
    )
```

**Comment by Chaoba** [comment]
> @Chaoba does it actually solve underlying issue? will it work now if using Style DSL:
> 
> ```
>  mapView.getMapboxMap().loadStyle(
>       styleExtension = style("") {
>         +rasterSource("osm"){
>           tileSet(tileSet)
>           tileSize(256)
>         }
>         +rasterLayer(LAYER_ID, "osm") {
> 
>         }
>       }
>     )
> ```

@kiryldz Yes, please see the attached screenshot

**Review by tobrun — COMMENTED** [review]
Would it be possible to not even require a string to be passed in?

```
 mapView.getMapboxMap().loadStyle(
      styleExtension = style() {
        +rasterSource("osm"){
          tileSet(tileSet)
          tileSize(256)
        }
        +rasterLayer(LAYER_ID, "osm") {

        }
      }
    )
```

File: sdk/src/main/java/com/mapbox/maps/MapboxMap.kt [hunk 2]
```diff
@@ -89,6 +89,8 @@ class MapboxMap internal constructor(
    * - **`file://...`**:
    * loads the style from a file path. This is used to load a style from disk.
    *
+   * Will load an empty json `{}` if the styleUri is empty.
```
Comment (pengdev) [inline sdk/src/main/java/com/mapbox/maps/MapboxMap.kt:92]: nit: we should add the same documentation to the `StyleExtension#StyleUri`

File: sdk/src/main/java/com/mapbox/maps/MapboxMap.kt [hunk 3]
```diff
@@ -99,7 +101,11 @@ class MapboxMap internal constructor(
     onMapLoadErrorListener: OnMapLoadErrorListener? = null
   ) {
     initializeStyleLoad(onStyleLoaded, onMapLoadErrorListener)
-    nativeMapWeakRef.call { (this as StyleManagerInterface).styleURI = styleUri }
+    if (styleUri.isEmpty()) {
+      nativeMapWeakRef.call { (this as StyleManagerInterface).styleJSON = "{}" }
```
Comment (alexander-kulikovskii) [inline sdk/src/main/java/com/mapbox/maps/MapboxMap.kt:105]:
```suggestion
      nativeMapWeakRef.call { (this as StyleManagerInterface).styleJSON = EMPTY_JSON_STYLE }
```
and add 
```
EMPTY_JSON_STYLE = "{}"
```

**Comment by Chaoba** [comment]
> Would it be possible to not even require a string to be passed in?
> 
> ```
>  mapView.getMapboxMap().loadStyle(
>       styleExtension = style() {
>         +rasterSource("osm"){
>           tileSet(tileSet)
>           tileSize(256)
>         }
>         +rasterLayer(LAYER_ID, "osm") {
> 
>         }
>       }
>     )
> ```

Add a default empty string for it and we can now use it in this way:
```
mapView.getMapboxMap().loadStyle(
      styleExtension = style {
        +rasterSource("osm"){
          tileSet(tileSet)
          tileSize(256)
        }
        +rasterLayer(LAYER_ID, "osm") {

        }
      }
    )
```

**Review by tobrun — APPROVED** [review]
Clean solution with just having `style { }`
LGTM, :shipit:

**Review by kiryldz — APPROVED** [review]

File: sdk/src/main/java/com/mapbox/maps/MapboxMap.kt (hunk 1 above)
Comment (yunikkk) [reply sdk/src/main/java/com/mapbox/maps/MapboxMap.kt:108]: Ok!

**Pull request merged**